packages/leancode_flutter_svg_adaptive_loader/lib/src/leancode_flutter_svg_adaptive_loader.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:vector_graphics/vector_graphics.dart';
+import 'package:vector_graphics_compiler/vector_graphics_compiler.dart' as vg;
+
+/// A [BytesLoader] able to distinguish xml-based svg from compiled binary form,
+/// by looking for '<svg' pattern at 1000 of first characters.
+/// Using logic from [AssetBytesLoader] for binaries and [SvgAssetLoader]
+/// for standard xml-based format.
+class LeancodeFlutterSvgAdaptiveLoader extends BytesLoader {
+  /// Creates a [BytesLoader]
+  const LeancodeFlutterSvgAdaptiveLoader(
+    this.assetName, {
+    this.packageName,
+    this.assetBundle,
+  });
+
+  /// The name of the asset to load.
+  final String assetName;
+
+  /// The package name to load from, if any.
+  final String? packageName;
+
+  /// The asset bundle to use.
+  ///
+  /// If unspecified, [DefaultAssetBundle.of] the current context will be used.
+  final AssetBundle? assetBundle;
+
+  AssetBundle _resolveBundle(BuildContext? context) {
+    if (assetBundle != null) {
+      return assetBundle!;
+    }
+    if (context != null) {
+      return DefaultAssetBundle.of(context);
+    }
+    return rootBundle;
+  }
+
+  Future<ByteData> _prepareMessage(BuildContext? context) {
+    return _resolveBundle(context).load(
+      packageName == null ? assetName : 'packages/$packageName/$assetName',
+    );
+  }
+
+  String _decode(ByteData? message, [int? lenght]) =>
+      utf8.decode(message!.buffer.asUint8List(0, lenght), allowMalformed: true);
+
+  /// This method intentionally avoids using `await` to avoid unnecessary event
+  /// loop turns. This is meant to to help tests in particular.
+  @override
+  Future<ByteData> loadBytes(BuildContext? context) {
+    return _prepareMessage(context).then((message) {
+      // Dart encodes strings in UTF-16 so one codepoint is 2 bytes
+      final first1000Chars = _decode(message, min(2000, message.lengthInBytes));
+
+      if (first1000Chars.contains('<svg')) {
+        return svg.cache.putIfAbsent(
+          cacheKey(context),
+          () => _loadSvg(context, message),
+        );
+      } else {
+        return message;
+      }
+    });
+  }
+
+  Future<ByteData> _loadSvg(
+    BuildContext? context,
+    ByteData? message,
+  ) {
+    return compute(
+      (message) {
+        return vg
+            .encodeSvg(
+              xml: _decode(message),
+              debugName: 'Svg loader',
+              enableClippingOptimizer: false,
+              enableMaskingOptimizer: false,
+              enableOverdrawOptimizer: false,
+            )
+            .buffer
+            .asByteData();
+      },
+      message,
+      debugLabel: 'Load Bytes',
+    );
+  }
+
+  @override
+  Object cacheKey(BuildContext? context) {
+    return _AssetByteLoaderCacheKey(
+      assetName,
+      packageName,
+      _resolveBundle(context),
+    );
+  }
+
+  @override
+  String toString() =>
+      'VectorGraphicAsset(${packageName != null ? '$packageName/' : ''}$assetName)';
```

Comment:
This name was used in flutter_svg code for both `AssetBytesLoader` and `SvgAssetLoader`, since our class is basically a copy of those with condition based on type I left it to be `VectorGraphicAsset`. Would you still recommend to rename it?